calculator.rb

```diff
@@ -0,0 +1,64 @@
+class Calculator
+  def initialize(overrides = {})
+    @calculate_sum = overrides.fetch(:calculate_sum) { Sum.new }
+    @calculate_subtraction = overrides.fetch(:calculate_subtraction) { Subtraction.new }
+    @calculate_multiplication = overrides.fetch(:calculate_multiplication) { Multiplication.new }
+    @calculate_division = overrides.fetch(:calculate_division) { Division.new }
+  end
+
+  def sum(number_1, number_2)
+    @calculate_sum.sum(number_1, number_2)
+  end
+
+  def subtraction(number_1, number_2)
+    @calculate_subtraction.subtraction(number_1, number_2)
+  end
+
+  def multiplication(number_1, number_2)
+    @calculate_multiplication.multiplication(number_1, number_2)
+  end
+
+  def division(number_1, number_2)
+    @calculate_division.division(number_1, number_2)
+  end
+
+  def calculate(operator, number_1, number_2)
+    case operator
+    when '+'
+      sum(number_1, number_2)
+    when '-'
+      subtraction(number_1, number_2)
+    when '*'
+      multiplication(number_1, number_2)
+    when '/'
+      division(number_1, number_2)
+    else
+      'O operador digitado é inválido.'
+    end
+  end
+end
+
+class Sum
+  def sum(number_1, number_2)
+    number_1 + number_2
+  end
+end
+
+class Subtraction
+  def subtraction(number_1, number_2)
```

Comment:
Uma dica caso queira evitar essa repetição do termo `sum` é padronizar o método da ação nas classes de operações matemáticas.
Por exemplo:
Ao invés de ter os métodos com os mesmos nomes das classes, você poderia ter algo do tipo:
```ruby
class Sum
  def calculate(number_1, number_2)
    number_1 - number_2
  end
end
```
E seguir o mesmo padrão para as outras classes também.

Aí, quando vc for usar as classes de operações na `Calculator`, você chamar as intancias delas com o nome da classe mesmo, que ficaria assim:

```ruby
class Calculator
  def initialize(*dependencias*)
    @sum = Sum.new
    @multiplication = Multiplication.new
  end

  def sum(number_1, number_2)
    @sum.calculate(number_1, number_2)
  end

  def multiplication(number_1, number_2)
    @multiplication.calculate(number_1, number_2)
  end
  ...
end
```

Assim, `sum.calculate` fica mais legível para quem estiver vendo o seu código.

Reply:
Muito obrigada pela sugestão e explicação, Toninho, ficou bem mais legível mesmo =D